src/main/java/com/github/kumaraman21/intellijbehave/parser/StoryFile.java

```diff
@@ -48,34 +45,14 @@ public FileType getFileType() {
     @NotNull
     public List<JBehaveStep> getSteps() {
 
-        List<ASTNode> stepNodes = newArrayList();
-
-        for (PsiElement scenario : getScenarios()) {
-            ASTNode[] stepNodesOfScenario = scenario.getNode().getChildren(STEPS_TOKEN_SET);
-            stepNodes.addAll(asList(stepNodesOfScenario));
-        }
-
-        return transform(stepNodes, new NodeToStepPsiElement());
-    }
-
-    @NotNull
-    private List<PsiElement> getScenarios() {
-        PsiElement story = getStory();
-        if (story == null) {
-            return newArrayList();
-        }
-
-        ASTNode[] scenarioNodes = story.getNode().getChildren(TokenSet.create(SCENARIO));
-        return transform(asList(scenarioNodes), new NodeToPsiElement());
-    }
-
-    private PsiElement getStory() {
-        ASTNode[] storyNodes = this.getNode().getChildren(TokenSet.create(STORY));
-
-        if (storyNodes.length > 0) {
-            return storyNodes[0].getPsi();
-        }
-
-        return null;
+        return Stream.of(this.getNode().getChildren(TokenSet.create(STORY)))
```

Comment:
I would split this stream in three parts again for readability, just like the methods